lib/date_time/date_time_core.c

```diff
@@ -25,10 +25,14 @@ BUILD_ASSERT(CONFIG_DATE_TIME_TOO_OLD_SECONDS <= CONFIG_DATE_TIME_UPDATE_INTERVA
 
 #define DATE_TIME_EVT_TYPE_PREVIOUS 0xFF
 
-static K_SEM_DEFINE(time_fetch_sem, 0, 1);
+K_THREAD_STACK_DEFINE(date_time_stack, CONFIG_DATE_TIME_THREAD_STACK_SIZE);
+struct k_work_q date_time_work_q;
 
-static void date_time_handler(struct k_work *work);
-static K_WORK_DELAYABLE_DEFINE(time_work, date_time_handler);
+static void date_time_update_work_fn(struct k_work *work);
+static K_WORK_DELAYABLE_DEFINE(date_time_update_work, date_time_update_work_fn);
+
+static void date_time_update_manual_work_fn(struct k_work *work);
+static K_WORK_DEFINE(date_time_update_manual_work, date_time_update_manual_work_fn);
```

Comment:
Not a big deal, but it came to my mind if it would be possible to define two works using the same `date_time_update_work_fn` function.

Reply:
Good point! Seems to work.